storage/innobase/btr/btr0cur.cc

```diff
@@ -3261,9 +3264,18 @@ btr_cur_ins_lock_and_undo(
 				index, thr, mtr, &prdt);
 			*inherit = false;
 		} else {
+#ifdef WITH_WSREP
+		if (!index->is_clust() && (index->type & DICT_UNIQUE) &&
+                    wsrep_thd_is_BF(thr_get_trx(thr)->mysql_thd, false)) {
+                  thr_get_trx(thr)->wsrep_UK_scan = true;
```

Comment:
Name of variable is confusing, wsrep_skip_gap_locks would be more descriptive name.

Reply:
nomenclature is very important, I picked wsrep_UK_scan flag just to "tell the world" that this transaction is now performing scanning along unique secondary index, later execution phase or other transactions, would use this information for relaxing gap locking, if need be. So this was the reasoning for the current name, but sure it can be changed though.